---
title: "Rice-GWAS-main"
author: "Aidan Angus-Henry"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
library(tidyverse)
library(statgenGWAS)
```

load some data from previous analysis

```{r}
load("../../Rice-PCA-SNPs/output/data_from_SNP_lab.Rdata")
```

Join the PCs, population assignments, and phenotypes

```{r}
pheno.geno.pca.pop <- left_join(geno.pca.pop, data.pheno, by=c("ID" = "NSFTVID"))

#get rid of spaces in the phenotype
colnames(pheno.geno.pca.pop) <- make.names(colnames(pheno.geno.pca.pop))

head(pheno.geno.pca.pop)
```


Explore a phenotype: I chose `Seed.length`

```{r}
pl <- pheno.geno.pca.pop %>% ggplot(mapping=aes(x=Seed.length))
pl <- pl + labs(title="Histogram of Seed.length")
pl <- pl + geom_histogram(fill="darkblue")
pl

pl2 <- pheno.geno.pca.pop %>% ggplot(mapping=aes(x=Seed.length))
pl2 <- pl2 + labs(title="Seed.length faceted by assigned population")
pl2 <- pl2 + facet_wrap(facets= ~ assignedPop )
pl2 <- pl2 + geom_histogram(fill="blue")
pl2

pl3 <- pheno.geno.pca.pop %>% ggplot(mapping=aes(x=Seed.length))
pl3 <- pl3 + labs(title="Seed.length faceted by Region")
pl3 <- pl3 + facet_wrap(facets= ~ Region )
pl3 <- pl3 + geom_histogram(fill="darkgreen")
pl3
```

```{r}
# Standard error of the mean function
sem <- function(x, na.rm=TRUE) {
  if(na.rm) x <- na.omit(x)
  sd(x)/sqrt(length(x))
}

pheno.geno.pca.pop %>% group_by(Region) %>% 
  summarize(mean.seed.l=mean(Seed.length,na.rm=T),
            sem.seed.l=sem(Seed.length)
  ) %>%
  arrange(desc(mean.seed.l))
```

perform an ANOVA to investigate whether or not the differences of the mean seed lengths for different regions are significant.

```{r}
aovsl <- aov(Seed.length ~ Region, data=pheno.geno.pca.pop)
summary(aovsl)
```

Through the ANOVA above, we can conclude that Seed length varies significantly by region.

Now it would be good to check if mean seed length varies significantly by fastStructure population assignment.

```{r}
pheno.geno.pca.pop %>% group_by(assignedPop) %>% summarize(mean.Seed.length=mean(Seed.length, na.rm = T), sem.Seed.length=sem(Seed.length)) %>%
 arrange(desc(mean.Seed.length))
```

```{r}
aov2 <- aov(Seed.length ~ assignedPop, data=pheno.geno.pca.pop)
summary(aov2)
```

This analysis reveals that seed length does vary significantly by assigned population, suggesting that population structure could be a problem in a future GWAS

### Time to prepare GWAS data

Load Genotype data
```{r}
Sys.setenv(VROOM_CONNECTION_SIZE="500000") # needed because the lines in this file are long.

data.geno <- read_csv("../../Rice-PCA-SNPs/input/Rice_44K_genotypes.csv.gz",
                      na=c("NA","00"))  %>%
  rename(ID=`...1`, `6_17160794` = `6_17160794...22252`) %>% 
  select(-`6_17160794...22253`)
```

Transform ID data into rownames (for statgenGWAS)
```{r}
data.geno <- data.geno %>% as.data.frame() %>% column_to_rownames("ID")
head(data.geno[,1:10])
```


```{r}
data.map <- data.frame(SNP=colnames(data.geno)) # create the object
head(data.map)
```

create the data.map object for GWAS
```{r}
data.map <- data.map %>%
  separate(SNP, into=c("chr", "pos"), sep="_", remove=FALSE, convert=TRUE ) %>%
  column_to_rownames("SNP")

head(data.map)
```

Create phenotype data
```{r}
data.pheno.small <- data.pheno %>%
  set_names(make.names(colnames(.))) %>%
  rename(genotype=NSFTVID) %>%
  select(genotype, where(is.numeric)) %>%
  as.data.frame() # for GWAS, need data frames, not tibbles ;()

head(data.pheno.small)
```

Lastly, create data frame of covariates
```{r}
data.cv <- geno.pca.pop %>%
  as.data.frame() %>%
  column_to_rownames("ID")
```


Create the gdata object
```{r}
gData.rice <- createGData(geno=data.geno, map = data.map, pheno = data.pheno.small, covar = data.cv)
```

Recode the genotype data (will look into SNP imputation in the future)
```{r}
gData.rice.recode <- gData.rice %>% codeMarkers(verbose = TRUE)
```

Create a kinship matrix to help correct for pop. structure
```{r}
data.kinship <- kinship(gData.rice.recode$markers)
```

